Sources/TuistGenerator/Graph/TargetNode.swift

```diff
@@ -59,14 +59,16 @@ class TargetNode: GraphNode {
             throw GraphLoadingError.targetNotFound(name, path)
         }
 
-        let dependencies: [GraphNode] = try target.dependencies.map {
-            try node(for: $0,
-                     path: path,
-                     name: name,
-                     platform: target.platform,
-                     cache: cache,
-                     circularDetector: circularDetector,
-                     modelLoader: modelLoader)
+        let dependencies: [GraphNode] = try target.dependencies.flatMap { dependency in
+            try target.platform.map { platform in
```

Comment:
If the dependency is also a multi-platform target, I don't think we need to add an entry per platform.